### ISSUE TITLE
[Feature Request] All day due date

### ISSUE THREAD

**Issue by monsieurhannes** [nextcloud/deck#1157]
Hi!
It would be great to be able to just select a date without having to define a time. I know I could just pick 23.55 and I have covered the whole day, but it would simplify the process.
A check box would do it for me, like in the calendar app:
![bild](https://user-images.githubusercontent.com/20651440/62213955-5dba2700-b3a4-11e9-8347-977de9045603.png)

Subsequently it would be great to show "today" in the card instead of "in 7 hours". But that's a minor thing...

**Comment by juliushaertl** [comment]
Makes sense to me :+1:

**Comment by jakobroehrl** [comment]
I think with the new VUE implementation this is solved, because 00:00 is the default time:

1. 
![grafik](https://user-images.githubusercontent.com/24530680/67361176-2c4a9500-f568-11e9-8639-6fed214e5ebf.png)

2. 
![grafik](https://user-images.githubusercontent.com/24530680/67361254-58feac80-f568-11e9-9caa-116a9064aa29.png)

3. 
![grafik](https://user-images.githubusercontent.com/24530680/67361278-6d42a980-f568-11e9-94b9-0e7accaef3b9.png)

4. 
![grafik](https://user-images.githubusercontent.com/24530680/67361298-792e6b80-f568-11e9-8793-a4ad8e89b9d1.png)

**Comment by juliushaertl** [comment]
Not sure, a task being due on 00:00 is still different than just being due on a specific date.

**Comment by monsieurhannes** [comment]
I think especially if there is a future implementation into the calendar app it would be good to differentiate. But I don't know how your thoughts are about that.
Still very good to have a default on 00:00!

**Comment by sunjam** [comment]
I was looking for an `All Day` option, like is available on Tasks app.

**Comment by roboticsalign** [comment]
> Not sure, a task being due on 00:00 is still different than just being due on a specific date.

I agree and would like to support this enhancement. I think, there are two relevant differences between 00:00 and all day:

1. The due date for an all day event is at the End of the day (~11:59pm), 00:00 is overdue before the work started: 
![Screenshot_20210107-100445_Chrome](https://user-images.githubusercontent.com/23562360/103873513-35257900-50d0-11eb-81ed-6e76f2acea2d.jpg)

2. A concrete time corresponds to a timezone and implies a worldwide unique date/time combination. An all day event could/should consider the timezone of the current user. If I create an all day event in UTC and move to a location in UTC-6, the due date should be shown as end of UTC-6.

What do you think?

**Comment by jakobroehrl** [comment]
> 
> 
> > Not sure, a task being due on 00:00 is still different than just being due on a specific date.
> 
> I agree and would like to support this enhancement. I think, there are two relevant differences between 00:00 and all day:
> 
>     1. The due date for an all day event is at the End of the day (~11:59pm), 00:00 is overdue before the work started:
>        ![Screenshot_20210107-100445_Chrome](https://user-images.githubusercontent.com/23562360/103873513-35257900-50d0-11eb-81ed-6e76f2acea2d.jpg)
> 
>     2. A concrete time corresponds to a timezone and implies a worldwide unique date/time combination. An all day event could/should consider the timezone of the current user. If I create an all day event in UTC and move to a location in UTC-6, the due date should be shown as end of UTC-6.
> 
> 
> What do you think?

I tried to solve it here like this: https://github.com/nextcloud/deck/pull/2332#pullrequestreview-509320938

**Comment by roboticsalign** [comment]
> > > Not sure, a task being due on 00:00 is still different than just being due on a specific date.
> > 
> > 
> > I agree and would like to support this enhancement. I think, there are two relevant differences between 00:00 and all day:
> > ```
> > 1. The due date for an all day event is at the End of the day (~11:59pm), 00:00 is overdue before the work started:
> >    ![Screenshot_20210107-100445_Chrome](https://user-images.githubusercontent.com/23562360/103873513-35257900-50d0-11eb-81ed-6e76f2acea2d.jpg)
> > 
> > 2. A concrete time corresponds to a timezone and implies a worldwide unique date/time combination. An all day event could/should consider the timezone of the current user. If I create an all day event in UTC and move to a location in UTC-6, the due date should be shown as end of UTC-6.
> > ```
> > 
> > 
> > What do you think?
> 
> I tried to solve it here like this: [#2332 (review)](https://github.com/nextcloud/deck/pull/2332#pullrequestreview-509320938)

I think, this would be a quick win. Later, we can try to identify a possible solution, that fits with the needs of calendar apps.

**Comment by danir-de** [comment]
Didn't see any related developments on this, what is the current status? It's currently a deal breaker for us.

**Comment by MartinH-open** [comment]
![image](https://user-images.githubusercontent.com/1383111/181994516-1b1b71b7-aa5f-4ae5-890e-399ade6207ef.png)
Still not solved in a consistent way. 
The Due Day selections: today, tomorrow, next week, next month set a time of 23:59 for a specific date.
But a day selection on the calendar display sets a default time  of 00:00 for the that day.
Instead best would be to set a "all day" due time as default. (This is more a time period - depending on what time zone you might consider this date; date might have passed for you - but not for others elsewhere.)
This default of 23:59 or 00:00 is unfortunate for the calendar integration because this would display at the far bottom or at the very top of the day display of time-slots for that day. Best would be the "all-day" setting as default or if not doable now , better select mid-day (12:00) noon time as default using my current time zone.

**Comment by poltpolt** [comment]
another problem with lack of all-day is that in calendar view you get spammed by 00:00 as the prefix to all entries
![obraz](https://user-images.githubusercontent.com/9408405/206435862-46986f77-f7ce-4969-bdbd-ae64fe71a4d6.png)

**Comment by That-Dude1111** [comment]
Any updates on this feature? I would love this feature with the ultimate hope that I can view my deck tasks in the Calendar app at the top of each day.
